common/io/src/kseqpp_fasta_parser.cpp

```diff
@@ -23,17 +23,21 @@ namespace claragenomics
 namespace io
 {
 
-FastaParserKseqpp::FastaParserKseqpp(const std::string& fasta_file)
+FastaParserKseqpp::FastaParserKseqpp(const std::string& fasta_file, int min_sequencece_length)
 {
     klibpp::KSeq record;
     klibpp::SeqStreamIn iss(fasta_file.data());
     std::vector<FastaSequence> seqs;
     int total_len = 0;
     while (iss >> record)
     {
-        FastaSequence seq = {record.name, record.seq};
-        total_len += record.seq.size();
-        reads_.emplace_back(seq);
+        FastaSequence seq    = {record.name, record.seq};
+        auto sequence_length = record.seq.size();
+        if (sequence_length >= min_sequencece_length)
+        {
+            total_len += sequence_length;
+            reads_.emplace_back(seq);
```

Comment:
Consider to replace `reads_.emplace_back(seq);` by `reads_.push_back(std::move(seq));` to save a copy from the `seq` object.

The idea of `emplace_back(list of arguments)` is to construct an object directly in the container, by forwarding the `list of arguments` to the constructor of the element class internally.
Since you pass the`seq` object (or more precisely an lvalue reference to it), the constructor used internally will be the copy constructor of FastaSequence. This means it is equivalent to `reads_.push_back(seq)`.
`emplace_back()` would be helpful if FastaSequence had a constructor `FastaSequence(std::string const& name, std::string const& seq)`. Then you could call `reads_.emplace_back(record.name, record.seq);` and avoid creating the `seq` object altogether.

With the temporary `seq` object, the only way to remove the copy into `reads_` is explicitly tell the compiler to move from the object in the call with either `push_back(std::move(seq));` or `emplace_back(std::move(seq));`. All STL containers have an overload `push_back(T&& x)`, so there is again no difference between `push_back` and `emplace_back` here.

Reply:
Thanks for this. I applied the change.